luda-editor/client/src/pages/graphic_asset_manage_page/mod.rs

```diff
@@ -0,0 +1,213 @@
+mod auto_column_list;
+mod cg_list;
+mod cg_viewer;
+mod image_list;
+mod image_viewer;
+mod side_bar;
+mod top_bar;
+mod upload_asset;
+
+use crate::{
+    app::notification::{self, Notification},
+    pages::graphic_asset_manage_page::{
+        cg_list::CgList,
+        cg_viewer::CgViewer,
+        image_list::ImageList,
+        image_viewer::ImageViewer,
+        side_bar::SideBar,
+        top_bar::TopBar,
+        upload_asset::{add_new_image, upload_file},
+    },
+};
+use futures::join;
+use namui::prelude::*;
+use namui_prebuilt::{simple_rect, table::hooks::*};
+use rpc::data::{CgFile, ImageWithLabels};
+use std::ops::Deref;
+
+static TAB_ATOM: Atom<Tab> = Atom::uninitialized_new();
+static IMAGES_ATOM: Atom<Vec<ImageWithLabels>> = Atom::uninitialized_new();
+static CG_FILES_ATOM: Atom<Vec<CgFile>> = Atom::uninitialized_new();
+static SELECTED_ASSET_ATOM: Atom<Option<SelectedAsset>> = Atom::uninitialized_new();
+
+#[namui::component]
+pub struct GraphicAssetManagePage {
+    pub wh: Wh<Px>,
+    pub project_id: Uuid,
+}
+
+impl Component for GraphicAssetManagePage {
+    fn render(self, ctx: &RenderCtx) -> RenderDone {
+        const TOP_BAR_HEIGHT: Px = px(48.0);
+        const SIDE_BAR_WIDTH: Px = px(192.0);
+
+        let Self { wh, project_id } = self;
+
+        let project_id = ctx.track_eq(&project_id);
+        let (tab, _set_tab) = ctx.atom_init(&TAB_ATOM, || Tab::Image);
+        let (_, _) = ctx.atom_init(&IMAGES_ATOM, Vec::new);
+        let (_, _) = ctx.atom_init(&CG_FILES_ATOM, Vec::new);
```

Comment:
I guess you don't have to use ctx.atom_init? Can't you init without ctx? I guess atom itself have init fn.

Reply:
Yes, there is ` pub fn init(&self, init: T) -> Result<()>` for Atom.
That will looks
```rust
ctx.effect("Init IMAGES_ATOM and CG_FILES_ATOM", || {
    let _ = IMAGES_ATOM.init(Vec::new());
    let _ = CG_FILES_ATOM.init(Vec::new());
});
```

Reply:
Fixed with af6cca3

Reply:
effect is not good idea. it looks very tiresome.

Reply:
Any suggestion?

Reply:
atom init without parenthesis?

Reply:
I don't know what you mean.
Can you show me the code?

Reply:
```
        let _ = ctx.atom_init(&IMAGES_ATOM, Vec::new);
        let _ = ctx.atom_init(&CG_FILES_ATOM, Vec::new);
```

Reply:
Fixed with 84a27e3